# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/nsExternalHelperAppService.cpp
# Commit: a5eab654b0eb
# Full Hash: a5eab654b0eb048549ab4828e20c20f46bea80c6
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2021-05-22 03:53:20
# Description:
#   Bug 1712275 - avoid crashing on browsingcontext-less downloads for which we ask the user what to do with the file, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115714
# ==============================================================================

diff -r b9b415f1038e -r a5eab654b0eb uriloader/exthandler/nsExternalHelperAppService.cpp
--- a/uriloader/exthandler/nsExternalHelperAppService.cpp	Sat May 22 00:29:53 2021 +0300
+++ b/uriloader/exthandler/nsExternalHelperAppService.cpp	Fri May 21 21:59:59 2021 +0000
@@ -1449,7 +1449,7 @@
 
 NS_IMETHODIMP nsExternalAppHandler::GetBrowsingContextId(
     uint64_t* aBrowsingContextId) {
-  *aBrowsingContextId = mBrowsingContext->Id();
+  *aBrowsingContextId = mBrowsingContext ? mBrowsingContext->Id() : 0;
   return NS_OK;
 }
 
